tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,172 @@
+from tap_tester import runner, connections, menagerie
+from base import AsanaBase
+from dateutil import parser as parser
+
+
+class AsanaInterruptedSyncTest(AsanaBase):
+    """
+    Test to verify that if a sync is interrupted, then the next sync will continue
+    from the bookmarks and currently syncing stream.
+    """
+
+    def name(self):
+        return "tap_tester_asana_interrupted_sync_test"
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_sync_stream`.
+                  The next sync job kicks off and the tap picks back up on that
+                  `currently_sync_stream` stream.
+
+        Expected State Structure:
+            {
+                "currently_sync_stream": "stream_name",
+                "bookmarks": {
+                    "stream_1": {"bookmark_field": "2010-10-10T10:10:10.100000"},
+                    "stream_2": {"bookmark_field": "2010-10-10T10:10:10.100000"}
+                }
+            }
+
+        Test Cases:
+        - Verify an interrupted sync can resume based on the `currently_sync_stream` and
+            stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the
+            stream level bookmark are replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream
+            in the resuming sync.
+        """
+
+        self.start_date = self.get_properties()["start_date"]
```

Comment:
variable should be named `start_date` and not `self.start_date`

Reply:
No, this is required while creating the connection in the backend, to be exact at below statement,

`conn_id = connections.ensure_connection(self)`